src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer producer = type.equals(Bulldozer.class) ? new BulldozerProducer()
+                : type.equals(Excavator.class) ? new ExcavatorProducer()
+                : type.equals(Truck.class) ? new TruckProducer()
+                : null;
```

Comment:
its better not to neste ternary expressions, it makes code less readable. rewrite with regular if-else